src/main/java/hudson/plugins/gradle/injection/GitScmListener.java

```diff
@@ -121,6 +128,25 @@ private static boolean shouldDisableMavenInjection(InjectionConfig config) {
         return config.isInjectMavenExtension();
     }
 
+    private static boolean mavenExtensionAlreadyApplied(InjectionConfig config, FilePath workspace) throws IOException, InterruptedException {
+        if (!config.isInjectMavenExtension()) {
+            return false;
+        }
```

Comment:
If injection of Maven extension is tuned off, we don't need to perform this check and simply leave things as they are